strawberryfields/engine.py

```diff
@@ -797,6 +797,6 @@ class BosonicEngine(LocalEngine):
 
     def _run_program(self, prog, **kwargs):
         # Custom Bosonic run code
-        applied, samples_dict, all_samples = self.backend.run_prog(prog, **kwargs)
+        applied, samples_dict = self.backend.run_prog(prog, **kwargs)
```

Comment:
Can you confirm we are not introducing a discrepancy in the `samples_dict` argument passed to `_combine_and_sort_samples()` below?

Reply:
The `samples_dict` here is actually `all_samples` renamed. They both contained more-or-less the same things, only that the _former_ overwrote the samples instead of storing them all. I chose to rename `all_samples` to `samples_dict` and simply remove the old `samples_dict`.

Reply:
Makes sense!